src/bridge/RoomCreation.ts

```diff
@@ -0,0 +1,96 @@
+import { IrcServer } from "../irc/IrcServer";
+import { IrcBridge } from "./IrcBridge";
+import { MatrixRoom, Intent } from "matrix-appservice-bridge";
+import { BridgeRequest } from "../models/BridgeRequest";
+import { RoomOrigin } from "../datastore/DataStore";
+
+interface TrackChannelOpts {
+    server: IrcServer;
+    ircChannel: string;
+    key?: string;
+    inviteList?: string[];
+    origin: RoomOrigin;
+    roomAliasName?: string;
+    intent?: Intent;
+}
+
+/**
+ * Track an IRC channel and create a room for it.
+ * @param ircBridge The ircBridge instance
+ * @param req The request that triggered the room creation
+ * @param opts Information about the room creation request.
+ */
+export async function trackChannelAndCreateRoom(ircBridge: IrcBridge, req: BridgeRequest, opts: TrackChannelOpts) {
+    const { server, ircChannel, key, inviteList, origin, roomAliasName } = opts;
+    const intent = opts.intent || ircBridge.getAppServiceBridge().getIntent();
+    const initialState: ({type: string; state_key: string; content: object})[] = [
+        {
+            type: "m.room.join_rules",
+            state_key: "",
+            content: {
+                join_rule: server.getJoinRule()
+            }
+        },
+        {
+            type: "m.room.history_visibility",
+            state_key: "",
+            content: {
+                history_visibility: "joined"
+            }
+        }
+    ];
+    if (server.areGroupsEnabled()) {
+        initialState.push({
+            type: "m.room.related_groups",
+            state_key: "",
+            content: {
+                groups: server.getGroupId(),
```

Comment:
This changed from an array to a string. Is that intended?

Reply:
No, oops. Good find.